pkg/netconfig/netconfig.go

```diff
@@ -0,0 +1,363 @@
+	}
+	return res
+}
+
+func AddIPListToIPSet(ipSetName string, content string) error {
+
+	addrs := prepareAddrs(content, false)
+
+	for i, addr := range addrs {
+		addrs[i] = "-A " + ipSetName + " " + addr
+	}
+	addrs = append(addrs, "quit\n")
+
+	err := DefaultNetUtilsCommandExecutor.ExecuteIPSetScript(strings.Join(addrs, "\n"))
+	if err != nil {
+		return fmt.Errorf("Error adding addresses to ipset %w\n", err)
+	}
+	fmt.Printf("Added %d entries to ipset %s.\n", len(addrs), ipSetName)
+	return nil
+}
+
+func UpdateIPSet(ipVersion, ipSetName, egressFilterList, defaultNetworkDevice string, blockIngress bool) error {
+	inetVersion := ""
+	if ipVersion == "6" {
+		inetVersion = "6"
+	}
+
+	defer func() {
+		fmt.Println("Clean-up")
+		err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("destroy", "tmpIPSet")
+		if err != nil {
+			fmt.Printf("Error cleaning-up ipsets %v\n", err)
+		}
+	}()
+
+	fmt.Printf("Creating temporary ipset with name \"%s\"...\n", "tmpIPSet")
+	err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("create", "tmpIPSet", "hash:net", "family", "inet"+inetVersion, "maxelem", ipSetsMaxLen)
+	if err != nil {
+		return fmt.Errorf("Error creating temporary ipset: %w\n", err)
+	}
+
+	fmt.Printf("Temporary ipset with name \"%s\" created successfully.\n", "tmpIPSet")
+
+	err = AddIPListToIPSet("tmpIPSet", egressFilterList)
+	if err != nil {
+		return fmt.Errorf("Error adding entries to temporary ipset: %w\n", err)
+	}
+	fmt.Println("Added entries into temporary ipset successfully.")
+
+	err = AddIPTablesLoggingRules(ipVersion, ipSetName, defaultNetworkDevice, blockIngress)
+	if err != nil {
+		return fmt.Errorf("Error adding iptables rules %w\n", err)
+	}
+
+	fmt.Printf("Swapping new ipset %s against old one %s\n", "tmpIPSet", ipSetName)
+	err = DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("swap", "tmpIPSet", ipSetName)
+	if err != nil {
+		return fmt.Errorf("Error swapping ipsets %w\n", err)
+	}
+
+	return nil
+}
+
+// blackholer
+
+func diff(new, old []string) (added, removed []string) {
+	newSet := make(map[string]bool)
+	oldSet := make(map[string]bool)
+
+	for _, s := range new {
+		newSet[s] = true
+	}
+
+	for _, s := range old {
+		oldSet[s] = true
+	}
+
+	for s := range newSet {
+		if !oldSet[s] {
+			added = append(added, s)
+		}
+	}
+
+	for s := range oldSet {
+		if !newSet[s] {
+			removed = append(removed, s)
+		}
+	}
+
+	return
+}
+
+func InitDummyDevice() error {
+	out, _ := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand("4", "link", "show")
+	if !strings.Contains(out.String(), " "+dummyDeviceName+": ") {
+		_, err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand("4", "link", "add", dummyDeviceName, "type", "dummy")
+		if err != nil {
+			return fmt.Errorf("Error creating dummy device: %v", err)
+		}
+		_, err = DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand("4", "link", "set", dummyDeviceName, "up")
+		if err != nil {
+			return fmt.Errorf("Error setting up dummy device: %v", err)
+		}
```

Comment:
Please consider adding a log entry after creating/initializing the dummy device.